app/models/gravatar.rb

```diff
@@ -0,0 +1,36 @@
+class Gravatar
+  attr_reader :email
+
+  def initialize(email, options = {})
+    raise ArgumentError, "Expected :email" unless email
+    @options = options || {}
+    @email = email
+  end
+
+  def image_url(options = {})
+    secure = options[:ssl] || options[:secure]
+    proto = "http#{secure ? 's' : ''}"
+    sub = secure ? "secure" : "www"
+
+    "#{proto}://#{sub}.gravatar.com/avatar/#{email_hash}#{extension_for_image(options)}#{query_for_image(options)}"
+  end
+
+  def email_hash(email = self.email)
+    Digest::MD5.hexdigest(email.downcase.strip)
+  end
+
+  def extension_for_image(options)
```

Comment:
We're only using two keys. Can we use kwargs instead of the options hash?

Reply:
My goal was to keep the original API. I'm ok with not doing that, I just wanted to keep the changes as minimal as possible. I can rewrite the Gravatar class to only support our usage